src/CommonProviderImplementation/AssemblyResolver.fs

```diff
@@ -7,211 +11,54 @@ open System.Reflection
-        | "FSharp.Core", "2.3.5.1" -> [fsharp31Portable47AssembliesPath]
-        | "FSharp.Core", "3.3.1.0" -> [fsharp31Portable7AssembliesPath]
-        | "FSharp.Data", _ -> fsharpDataPaths
-        | _, "2.0.5.0" -> [portable47AssembliesPath]
-        | _, _ -> []
-    let asm = 
-        folders |> List.tryPick (fun folder -> 
-          try
-            let assemblyPath = folder ++ (asmName.Name + ".dll")
-            if safeFileExists assemblyPath then
-                if reflectionOnly then nullToOption (Assembly.ReflectionOnlyLoadFrom assemblyPath)
-                else nullToOption (Assembly.LoadFrom assemblyPath)
-            else None 
-          with _ -> None)
-    match asm with 
-    | Some asm -> asm
-    | None -> 
-        if reflectionOnly then 
-          let redirectedName = AppDomain.CurrentDomain.ApplyPolicy(asmName.FullName)
-          Assembly.ReflectionOnlyLoad redirectedName
-        else null
+    [| yield Assembly.GetExecutingAssembly() 
+       for asm in Assembly.GetExecutingAssembly().GetReferencedAssemblies() do
+         let asm = try Assembly.Load(asm) with _ -> null
+         if asm <> null then 
+            yield asm |]
 
 let mutable private initialized = false    
 
 [<RequireQualifiedAccess>]
-type FSharpDataRuntimeVersion =
+type FSharpDataRuntimeInfo =
     | Net40
-    | Portable7 //net45+win8
-    | Portable47 //net45+wp8+win8
+    | Portable_47_7_259
     member x.SupportsLocalFileSystem = 
         match x with
         | Net40 -> true
-        | _ -> false
+        | Portable_47_7_259 -> false
+
 
 let init (cfg : TypeProviderConfig) = 
 
-    let init (_:SaveOptions) = 
-        WebRequest.DefaultWebProxy.Credentials <- CredentialCache.DefaultNetworkCredentials
-        AppDomain.CurrentDomain.add_AssemblyResolve(fun _ args -> getAssembly (AssemblyName args.Name) false [])
-        AppDomain.CurrentDomain.add_ReflectionOnlyAssemblyResolve(fun _ args -> getAssembly (AssemblyName args.Name) true [])
-        ProvidedTypes.ProvidedTypeDefinition.Logger := Some FSharp.Data.Runtime.IO.log
+    let bindingContext = cfg.GetTypeProviderBindingContext()
 
-    if not initialized then
-        initialized <- true
-        // the following parameter is just here to force System.Xml.Linq to load
-        init SaveOptions.None
-    
-    let useReflectionOnly = true
+    let version = 
+        let fsCore = bindingContext.TryGetFSharpCoreAssembly()
 
-    let runtimeAssembly = 
-        if useReflectionOnly then Assembly.ReflectionOnlyLoadFrom cfg.RuntimeAssembly
-        else Assembly.LoadFrom cfg.RuntimeAssembly
+        match fsCore with 
+        | Choice2Of2 _err -> FSharpDataRuntimeInfo.Net40
+        | Choice1Of2 asm -> 
+           let ver = asm.GetName().Version
+           match ver.Major, ver.Minor with 
+            | 4, _ -> FSharpDataRuntimeInfo.Net40 // 4.3.0.0, 4.3.1.0, 4.4.0.0
+            | _ -> FSharpDataRuntimeInfo.Portable_47_7_259
```

Comment:
Could this assumption break in the future and we get a portable profile with major version 4?